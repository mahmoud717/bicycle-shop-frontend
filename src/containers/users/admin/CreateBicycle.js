/* eslint-disable no-unused-expressions */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import axios from 'axios';
import { useHistory } from 'react-router-dom';

const CreateBicycle = () => {
  const history = useHistory();
  const [name, setName] = useState('');
  const [model, setModel] = useState('');
  const [image, setImage] = useState('');
  const [description, setDescription] = useState('');
  const handleSubmit = e => {
    e.preventDefault();
    axios.post('http://localhost:5000/api/v1/bicycles', {
      name,
      model,
      image_url: image,
      description,
    })
      .then(response => {
        !response.data.status && history.push(`/bicycles/${response.data.id}`);
      });
  };
  return (
    <div className="bicycle-form-container d-flex justify-content-center w-100 align-items-center">
      <form className="bicycle-form w-100" onSubmit={handleSubmit}>
        <h1 className="text-center">Create Bicycle</h1>
        <div className="form-group w-100">
          <label htmlFor="exampleFormControlInput1">Name</label>
          <input type="text" className="form-control" id="exampleFormControlInput1" placeholder="Product Name" value={name} onChange={e => setName(e.target.value)} required />
        </div>
        <div className="form-group">
          <label htmlFor="exampleFormControlInput2">Model</label>
          <input type="text" className="form-control" id="exampleFormControlInput2" placeholder="Product Model" value={model} onChange={e => setModel(e.target.value)} required />
        </div>
        <div className="form-group">
          <label htmlFor="exampleFormControlInput3">Image URL</label>
          <input type="text" className="form-control" id="exampleFormControlInput3" placeholder="Product Image link" value={image} onChange={e => setImage(e.target.value)} required />
        </div>
        <div className="form-group">
          <label htmlFor="exampleFormControlInput4">Description</label>
          <textarea className="form-control" id="exampleFormControlInput4" placeholder="Add product description" value={description} onChange={e => setDescription(e.target.value)} required />
        </div>
        <div className="text-center">
          <input type="submit" value="Create" className="btn text-center btn-primary" />
        </div>

      </form>
    </div>
  );
};

export default CreateBicycle;
