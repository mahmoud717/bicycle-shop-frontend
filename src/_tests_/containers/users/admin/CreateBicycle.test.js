import React from 'react';

import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CreateBicycle from '../../../../containers/users/admin/CreateBicycle';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <CreateBicycle />
  </Provider>,
);

describe('create bicycle Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the create bicycle container', () => {
    const page = component.find('.bicycle-form-container');
    expect(page).toMatchSnapshot();
  });
});
