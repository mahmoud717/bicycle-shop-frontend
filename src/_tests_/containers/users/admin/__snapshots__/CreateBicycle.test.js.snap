// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create bicycle Component renders the create bicycle container 1`] = `ShallowWrapper {}`;
